### DISCUSSION TITLE
Version 0.6.3

### DISCUSSION THREAD

**Discussion by IvyZX** [google/flax#2706] [General]
## What's Changed
* Add gfile api shim to remove tensorflow dependency for basic IO. by @chiamp in https://github.com/google/flax/pull/2586
* Remove Mypy type errors by @zaxtax in https://github.com/google/flax/pull/2594
* Attempt to fix pytype issue. by @levskaya in https://github.com/google/flax/pull/2628
* example/imagenet: use absolute path to locate the Flax root dir by @yhtang in https://github.com/google/flax/pull/2630
* Move Flax - The Sharp Bits ToC location by @8bitmp3 in https://github.com/google/flax/pull/2633
* Speed up Github Actions CI by @levskaya in https://github.com/google/flax/pull/2635
* Update requirements by @marcvanzee in https://github.com/google/flax/pull/2652
* Switch to Orbax for Flax single-checkpoint support under the hood. by @copybara-service in https://github.com/google/flax/pull/2637
* Added path discrepancy details in serialization errors by @chiamp in https://github.com/google/flax/pull/2632
* Fix get_type_hints again by @cgarciae in https://github.com/google/flax/pull/2654
* BatchNorm guide by @cgarciae in https://github.com/google/flax/pull/2536
* Generalize pool to handle multiple batch dimensions by @cgarciae in https://github.com/google/flax/pull/2591
* Improve transfer learning guide by @cgarciae in https://github.com/google/flax/pull/2595
* update docstrings and error messages in traverse_util.py. by @copybara-service in https://github.com/google/flax/pull/2666
* Use a different rng key for each batch element in DenseGeneral init by @j-towns in https://github.com/google/flax/pull/2665
* Added check for Mac M1 chip, when conducting serialization test. by @chiamp in https://github.com/google/flax/pull/2657
* Added explicit warning if flax.io is using default Python I/O operations by @chiamp in https://github.com/google/flax/pull/2625
* Check file or directory before removing checkpoints. by @IvyZX in https://github.com/google/flax/pull/2676
* updated extracting_intermediates in flax docs by @chiamp in https://github.com/google/flax/pull/2685
* updated model_surgery in flax_docs by @chiamp in https://github.com/google/flax/pull/2687
* updated getting_started in flax docs by @chiamp in https://github.com/google/flax/pull/2684
* Updated flax docs. by @chiamp in https://github.com/google/flax/pull/2667
* updated flax_basics in flax docs by @chiamp in https://github.com/google/flax/pull/2686
* Update python version support by @cgarciae in https://github.com/google/flax/pull/2682
* add mypy.ini placeholder. by @copybara-service in https://github.com/google/flax/pull/2693
* Release 0.6.3 by @IvyZX in https://github.com/google/flax/pull/2705

## New Contributors
* @yhtang made their first contribution in https://github.com/google/flax/pull/2630

**Full Changelog**: https://github.com/google/flax/compare/v0.6.2...v0.6.3

<hr /><em>This discussion was created from the release <a href='https://github.com/google/flax/releases/tag/v0.6.3'>Version 0.6.3</a>.</em>
